lib/main.js

```diff
@@ -841,6 +843,17 @@ exports.typeset = function (data,callback) {
   if (serverState == STATE.READY) StartQueue();
 }
 
+// main API, callback and promise compatible
+exports.typeset = function (data, callback) {
+    if (callback) cbTypeset(data, callback);
+    else return new Promise(function (resolve, reject) {
+        cbTypeset(data, function (output, input) {
+            if (output.errors) reject(output.errors);
```

Comment:
> I would have said no since `undefinedCharError` is false by default

I thought it was true, but I see that you are right, so I agree, no need to do anything about it.  But at least we did find out that the CHTML undefined character signals aren't currently being trapped.  :-)